lib/elixir/lib/keyword.ex

```diff
@@ -1316,4 +1316,87 @@ defmodule Keyword do
   def size(keywords) do
     length(keywords)
   end
+
+  @doc """
+  Filters `keywords` i.e. returns a keyword list containing
+  only the elements from `keywords` for which the function
+  `fun` returns a truthy value.
+
+  See also `reject/2` which discards all elements where the function
+  returns a truthy value.
+
+  ## Examples
+
+      iex> Keyword.filter([one: 1, two: 2, three: 3], fn {_key, val} -> rem(val, 2) == 1 end)
+      [one: 1, three: 3]
+
+  """
+  @doc since: "1.13.0"
+  @spec filter(t, ({key, value} -> as_boolean(term))) :: t
+  def filter(keywords, fun) when is_list(keywords) and is_function(fun, 1) do
+    do_filter(keywords, fun)
+  end
+
+  defp do_filter([], _fun), do: []
+
+  defp do_filter([element | vals], fun) do
+    if fun.(element) do
+      [element | do_filter(vals, fun)]
+    else
+      do_filter(vals, fun)
+    end
+  end
+
+  @doc """
+  Returns a keyword list excluding the elements from `keywords`
+  for which the function `fun` returns a truthy value.
+
+  See also `filter/2` which discards all elements where the function
+  returns a falsy value.
+
+  ## Examples
+
+      iex> Keyword.reject([one: 1, two: 2, three: 3], fn {_key, val} -> rem(val, 2) == 1 end)
+      [two: 2]
+
+  """
+  @doc since: "1.13.0"
+  @spec reject(t, ({key, value} -> as_boolean(term))) :: t
+  def reject(keywords, fun) when is_list(keywords) and is_function(fun, 1) do
+    do_reject(keywords, fun)
+  end
+
+  defp do_reject([], _fun), do: []
+
+  defp do_reject([element | vals], fun) do
+    if fun.(element) do
+      do_reject(vals, fun)
+    else
+      [element | do_reject(vals, fun)]
+    end
+  end
+
+  @doc """
+  Maps the function `fun` over all `{key, value}`-elements in `keywords`,
+  returning a keyword list with all the values replaced with
+  the result of the function.
```

Comment:
The Elixir stdlib docs are not consistent about this, but it does seem that the most common rendering of this is "key-value pairs", fully outside code blocks, whether or not the tuple literal is in the spec.